extra_files/fv3jedi_ufs_mod.F90

```diff
@@ -0,0 +1,585 @@
+         filename='model_configure', &
+         rc=rc)
+    
+    ! This call to read_input_nml() seems to be required
+    ! for CCPP.  However, it does not belong at this level
+    ! but should be handled inside the model itself
+    call read_input_nml()
+    call ESMF_LogWrite("done reading input nml", ESMF_LOGMSG_INFO)
+
+    ! Create the ESM component
+    self%esmComp = ESMF_GridCompCreate(name="esm", rc=rc)
+    esmf_err_abort(rc)
+
+    ! SetServices for the ESM component
+    call ESMF_GridCompSetServices(self%esmComp, esmSS, userRc=urc, rc=rc)
+    esmf_err_abort(rc)
+    esmf_err_abort(urc)
+
+   
+    ! Set ESM's Verbosity (High)  - 32513
+    call NUOPC_CompAttributeSet(self%esmComp, name="Verbosity", &
+         value="32513", rc=rc)
+    esmf_err_abort(rc)
+
+    
+    
+    ! Initialize the clock based on contents of model_configure
+    ! -------------------------------------------
+    call setUFSClock(self,startTime,stopTime)
+    call ESMF_GridCompSet(self%esmComp, clock=self%clock, rc=rc)
+
+    ! Create import and export states from
+    ! perspective of the exernal system:
+    !   toJedi is an IMPORT into Jedi and an EXPORT from ESM
+    !   fromJedi is an EXPORT from Jedi and an IMPORT into ESM
+
+    self%toJedi = ESMF_StateCreate(stateintent=ESMF_STATEINTENT_IMPORT, &
+         rc=rc)
+    esmf_err_abort(rc)
+
+     
+    self%fromJedi = ESMF_StateCreate(stateintent=ESMF_STATEINTENT_EXPORT, &
+         rc=rc)
+    esmf_err_abort(rc)
+
+     
+    call ESMF_LogWrite("Advertising export from ESM", ESMF_LOGMSG_INFO)
+    ! Advertise fields on the exportState, for data coming out of ESM component
+    ! Note--only certain fields are available. Check in GFS_surface_generic to see if they are filled
+    call NUOPC_Advertise(self%toJedi, &
+         StandardNames=(/ &
+                        "mean_down_lw_flx                     ", &   ! Example fields
+                        "mean_down_sw_flx                     ", &   ! Example fields
+                        "inst_zonal_wind_height10m            ", &   ! Example fields
+                        "inst_merid_wind_height10m            ", &   ! Example fields
+                        "inst_spec_humid_height_lowest        ", &   ! Example fields
+                        "inst_pres_height_lowest              ", &   ! Example fields
+                        "inst_height_lowest                   ", &   ! Example fields
+                        "inst_temp_height_lowest              ", &   ! Example fields
+                        "inst_merid_wind_height_lowest        ", &   ! Example fields
+                        "mean_zonal_moment_flx                ", &   ! Example fields
+                        "inst_down_sw_flx                     ", &   ! Example fields
+                        "inst_temp_height2m                   ", &   ! Example fields
+                        "inst_zonal_wind_height_lowest        "/), &   ! Example fields
```

Comment:
This list can be expanded with fields from other components (e.g., ocean, ice).  Do we need namespaces for these, or just a flat list?

Reply:
Fields advertised on ALL processors - if sharing is true then pointer will be shared and some processors will not have any data.  But empty ESMF_Fields will exist on all.  Without sharing - you could redistribute fields from subset of processors to all processors - so coupled state would be fully distributed across all processors.